packages/react-components/react-motion/library/src/factories/createMotionComponent.ts

```diff
@@ -94,12 +99,16 @@ export function createMotionComponent<MotionParams extends Record<string, Motion
 
       if (element) {
         const atoms = typeof value === 'function' ? value({ element, ...paramsRef.current }) : value;
-        onMotionStart();
 
         const handle = animateAtoms(element, atoms, { isReducedMotion: isReducedMotion() });
-
-        handle.setMotionEndCallbacks(onMotionFinish, onMotionCancel);
         handleRef.current = handle;
+        handle.setMotionEndCallbacks(onMotionFinish, onMotionCancel);
+
+        onMotionStart();
```

Comment:
Let's move it to the original position. It could be somehow a breaking change as now it's after it motion starts